Sprint-1/implement/sum.test.js

```diff
@@ -13,24 +13,48 @@ const sum = require("./sum.js");
 // Given an empty array
 // When passed to the sum function
 // Then it should return 0
-test.todo("given an empty array, returns 0")
+test("given an empty array, returns 0", () => {
+    expect(sum([])).toBe(0);
+});
 
 // Given an array with just one number
 // When passed to the sum function
 // Then it should return that number
+test("given an array with just one number, returns that number", () => {
+    expect(sum([42])).toBe(42);
+});
 
 // Given an array containing negative numbers
 // When passed to the sum function
 // Then it should still return the correct total sum
+test("given an array containing negative numbers, returns the correct total sum", () => {
+    expect(sum([-10, -20, 30])).toBe(0);
+});
 
 // Given an array with decimal/float numbers
 // When passed to the sum function
 // Then it should return the correct total sum
+test("given an array with decimal/float numbers, returns the correct total sum", () => {
+    expect(sum([1.5, 2.5, 3.5])).toBe(7.5);
```

Comment:
Decimal numbers in most programming languages (including JS) are internally represented in "floating point number" format. Floating point arithmetic is not exact. For example, the result of `46.5678 - 46 === 0.5678` is false because `46.5678 - 46` only yield a value that is very close to `0.5678`.

So `expect(sum([1.1, 1.1, 1.1])).toEqual(3.3);` would probably fail.

Can you find a more appropriate way to test a value (that involves decimal number calculations) for "equality?

Reply:
Hi,

I updated the test to use toBeCloseTo instead of toBe. This accounts for the potential small inaccuracies in floating point arithmetic when dealing with decimal numbers. The toBeCloseTo matcher allows us to specify a tolerance (in this case, 5 decimal places), ensuring that the result is close enough to the expected value despite any minor precision errors.

I believe this is a more appropriate approach. Please let me know if there is another way.

Thank you! 😊

Reply:
The `.toBeCloseTo()` is specific to Jest. An alternative is to use the "typical" approach: 
Consider two floating point numbers, A and B, are equal if their absolute difference is smaller than a tolerable number. For example,

`Math.abs(A - B) < 1e-10`

Reply:
Hi,

I’ve updated my tests to use the typical approach by comparing floating-point numbers with Math.abs(A - B) < 1e-10.

Since I’m still new to JavaScript, especially when it comes to writing test files, I initially found this concept a bit tricky to understand. However, after reviewing how it works, I think I’ve applied it correctly in my updated code.

I hope my current implementation is correct, but please let me know if there’s anything else I should adjust.

Thank you so much for your guidance! 😊